### DISCUSSION TITLE
Is there a Watcher component like Controller ?

### DISCUSSION THREAD

**Discussion by Sajarin-M** [react-hook-form/react-hook-form#9583] [General] [answered]
Is there a Watcher (or anything like) component which can be used like Controller but to watch and render which can avoid usage of `useWatch` in a nested component.
like : 

```
<Watcher
     control={control}
     name="fieldName"
     render={({value})=> {
          return /*...*/
    }}
/>

**Comment by Moshyfawn** [comment, marked as answer]
No, there's not. But you can easily create a `Watch` component like
```jsx
function Watch () {
  const data = useWatch()

  return data
} 
```

**Reply by Sajarin-M** [reply]
Thanks ❤️
I was just confirming if there isn't already such component before creating one

**Reply by Moshyfawn** [reply]
Yep, makes sense :P

The scope of this lib is to provide hooks for the user to easily build their forms. We have `Controller`, but that's about it for the component API RHF provides
